scripts/lint_jjb.py

```diff
@@ -80,6 +80,8 @@ def parse_jjb_file(in_dir, in_file):
                 _rc = 1
             if check_timed_trigger(item['job-template'], filename):
                 _rc = 1
+            if invalid_protocol(item['job-template'], filename):
```

Comment:
Needs a check in the job branch, and also the project branch. 


Many projects have a repo_url key, so that can use a check similar to the current implementation here. 

Some projects have lists of repos, each list item either has a `URL` or a `repo_url` key. I really wish we had some consistency, but I know that most of the inconsistency is my fault 😢

```./rpc_maas.yml-- project:
./rpc_maas.yml-    name: 'rpc-maas-whisperer'
./rpc_maas.yml-    series:
./rpc_maas.yml-      - all_branches:
./rpc_maas.yml-          branches: '.*'
./rpc_maas.yml:    repo:
./rpc_maas.yml-      - rpc-maas:
./rpc_maas.yml-          repo_url: 'https://github.com/rcbops/rpc-maas'
./rpc_maas.yml-    jobs:
./rpc_maas.yml-      - 'Pull-Request-Whisperer_{repo}'
./rpc_maas.yml-
```
or
```
./rpc_ceph.yml-- project:
./rpc_ceph.yml-    name: "rpc-ceph-re-release-pr"
./rpc_ceph.yml:    repo:
./rpc_ceph.yml-      - rpc-ceph:
./rpc_ceph.yml-          URL: "https://github.com/rcbops/rpc-ceph"
./rpc_ceph.yml-          BRANCH: "master"
./rpc_ceph.yml-    jobs:
./rpc_ceph.yml-      - 'RE-Release-PR_{repo}-{BRANCH}'
```
or
```
./rpc_uam.yml-- project:
./rpc_uam.yml:    name: 'rpc-uam-checkmarx'
./rpc_uam.yml-    jira_project_key: "FLEEK"
./rpc_uam.yml-    trigger:
./rpc_uam.yml-      - PM
./rpc_uam.yml-    scan_type:
./rpc_uam.yml-      - all
./rpc_uam.yml-      - pci
./rpc_uam.yml-    repo_name:
./rpc_uam.yml-      - rpc-uam:
./rpc_uam.yml-          repo_url: "git@github.com:rcbops/rpc-uam"
./rpc_uam.yml-          branch: master
./rpc_uam.yml-    jobs:
./rpc_uam.yml:      - '{trigger}-Checkmarx_{scan_type}-{repo_name}'
```

We really should get rid of all those different names/structures for the same thing, or drop jjb alltogether.

Reply:
I guess you can choose whether you want to code for the inconsistency, or fix the inconsistencies and have simpler code :)

Some docs may need updating if the param names are changed though.

Reply:
The feedback was helpful. I attempted to write a few routines in the lastest update to walk the different structures (dictionary in python) to pick out the keys and test for the appropriate protocol.